front/cypress/e2e/proposal_page_action.cy.ts

```diff
@@ -1,28 +1,59 @@
-import { randomString, randomEmail } from '../support/commands';
+import { randomEmail, randomString } from '../support/commands';
 
 describe('Initiative show page actions', () => {
   describe('not logged in', () => {
-    before(() => {
-      cy.visit('/initiatives/cleaning-the-sidewalks-party');
-      cy.get('#e2e-initiative-show');
-      cy.acceptCookies();
+    const initiativeTitle = randomString();
+    const initiativeContent = Math.random().toString(36);
+    let initiativeId: string;
+    let initiativeSlug: string;
+
+    beforeEach(() => {
+      cy.apiCreateInitiative({ initiativeTitle, initiativeContent })
+        .then((initiative) => {
+          initiativeId = initiative.body.data.id;
+          initiativeSlug = initiative.body.data.attributes.slug;
+        })
+        .then(() => {
+          cy.clearCookies();
+          cy.visit(`/initiatives/${initiativeSlug}`);
+          cy.get('#e2e-initiative-show');
+          cy.acceptCookies();
+        });
+    });
+
+    afterEach(() => {
+      cy.apiRemoveInitiative(initiativeId);
     });
 
     it('asks unauthorised users to log in or sign up before they vote', () => {
-      cy.wait(500);
-      cy.get(
-        '#e2e-initiative-vote-control #e2e-initiative-upvote-button'
-      ).click();
-      cy.get('#e2e-sign-up-in-modal');
+      cy.get('#e2e-initiative-upvote-button').should('exist');
+      cy.wait(2000);
```

Comment:
Ideally we don't use cy.wait() in tests, but after trying everything to get this test passing on CI, this seems to be the only consistent solution.